pkg/kubectl/cmd/log.go

```diff
@@ -164,74 +168,29 @@ func (o *LogsOptions) Validate() error {
 
 // RunLog retrieves a pod log
 func (o *LogsOptions) RunLog() error {
-	pod, err := o.Client.Pods(o.PodNamespace).Get(o.PodName)
+	mapper, typer := o.Factory.Object()
+	infos, err := resource.NewBuilder(mapper, typer, o.Factory.ClientMapperForCommand()).
+		NamespaceParam(o.Namespace).DefaultNamespace().
+		ResourceNames("pods", o.Name).
+		SingleResourceType().
+		Do().Infos()
 	if err != nil {
 		return err
 	}
-
-	// [-c CONTAINER]
-	container := o.ContainerName
-	if len(container) == 0 {
-		// [CONTAINER] (container as arg not flag) is supported as legacy behavior. See PR #10519 for more details.
-		if len(pod.Spec.Containers) != 1 {
-			podContainersNames := []string{}
-			for _, container := range pod.Spec.Containers {
-				podContainersNames = append(podContainersNames, container.Name)
-			}
-
-			return fmt.Errorf("Pod %s has the following containers: %s; please specify the container to print logs for with -c", pod.ObjectMeta.Name, strings.Join(podContainersNames, ", "))
```

Comment:
This check already happens in the server (https://github.com/kubernetes/kubernetes/blob/f5da178738bd1440c560c8f66b3b9adf1e342238/pkg/registry/pod/strategy.go#L233-L241) so I have removed it entirely from here.